src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Pattern patternData = Pattern.compile("^(3[01]|[12][0-9]|0?[1-9])\\.(1[012]|0?[1-9])"
```

Comment:
It's pretty expensive object, better to move it to class field so it will be created only once